python/tests/kat/t_active_health_check.py

```diff
@@ -0,0 +1,129 @@
+import time
+from typing import Generator, Tuple, Union
+
+from abstract_tests import AmbassadorTest, HealthCheckServer, Node
+from kat.harness import Query
+
+
+class ActiveHealthCheckTest(AmbassadorTest):
+    def init(self):
+        self.target = HealthCheckServer()
+
+    def config(self) -> Generator[Union[str, Tuple[Node, str]], None, None]:
+        yield self, self.format(
+            """
+---
+apiVersion: getambassador.io/v3alpha1
+kind: Mapping
+name:  {self.target.path.k8s}-health
+hostname: "*"
+prefix: /healthcheck/
+service: {self.target.path.fqdn}
+resolver: endpoint
+load_balancer:
+  policy: round_robin
+health_checks:
+- unhealthy_threshold: 1
+  interval: 1s
+  health_check:
+    http:
+      path: /healthcheck/actualcheck/
+"""
+        )  # The round robin load balancer is not necessary for the test but should help make the request distribution even across the pods
+
+    def queries(self):
+        yield Query(self.url("healthcheck/"), phase=1)  # Just making sure things are running
+        yield Query(self.url("ambassador/v0/diag/"), phase=1)
+
+        yield Query(
+            self.url("healthcheck/makeUnhealthy/"), phase=1
+        )  # the deployment has 5 pods. This will make one of them start returning errors
```

Comment:
Something needs to reset this at startup, otherwise running the test repeatedly won't work, as more and more pods get marked unheathly each run.  That would be unacceptable for locally running the tests as you hack on the code.

There are two ways that existing tests do such resets:
- Most of them call a "reset" endpoint in `requirements()` (example: `t_circuitbreaker.py`).  This might be problematic because we'd need to be sure to hit each Pod.
- `t_tracing.py`'s `Zipkin` does a thing with a nonce in an env-var, since there's no way to reset Zipkin. This might be the best option.

Reply:
Went with the nonce approach for this